### PR TITLE
wit/test: skip testing `HasBorrow` for anonymous types

### LINKED ISSUE #167: wit: logical flaw in `TestHasBorrow` function
Consider the following WIT:

```wit
package foo:bar;

world x {
  resource y;

  record r {
    y: borrow<y>
  }

  import f: func(a: list<r>);
}
```

The `TestHasBorrow` function returns an error for this case saying `testdata_test.go:493: has "borrow<": false != HasBorrow(td): true (list<r>)`

But `list<r>` does contain a `borrow` type, in record `r`'s field. I believe that simplify by asserting if the type WIT has `borrow<` syntax is not enough due to the existance of the anonymous types (e.g. `list`, `result`, `option`, etc.) in WIT.

## Discussion

For anonymous types like `list`, we will need to recursively go into the nested type to check if borrow type exists, this might be a bit hard to do as a unit test case, because you might need to re-implement the recursive logic for that. I am thinking about skipping the anonymous type for this case and adding specific, predefined test cases with known expected results.

### PR DESCRIPTION
**Pull request by Mossaka**
Closes #167

### CODE REVIEW

File: wit/testdata_test.go [hunk 1]
```diff
@@ -486,11 +486,15 @@ func TestHasBorrow(t *testing.T) {
...
 		t.Run(path, func(t *testing.T) {
 			for _, td := range res.TypeDefs {
-				wit := td.Kind.WIT(nil, "")
-				a := strings.Contains(wit, "borrow<")
-				b := HasBorrow(td)
-				if a != b {
-					t.Errorf("has \"borrow<\": %t != HasBorrow(td): %t (%s)", a, b, wit)
+				// skipping anonymous types
+				// see https://github.com/bytecodealliance/wasm-tools-go/issues/167
+				if td.Name != nil {
```
Comment (ydnar) [inline wit/testdata_test.go:491]: Can you change this to an early out instead of a nested if?

// Skip anonymous types here. This is tested elsewhere.

**Review by ydnar — COMMENTED** [review]
Thanks!

File: wit/testdata_test.go (hunk 1 above)
Comment (Mossaka) [reply wit/testdata_test.go]: Done.

File: wit/testdata_test.go [hunk 2]
```diff
@@ -499,6 +504,43 @@ func TestHasBorrow(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
+	testCases := []struct {
```
Comment (ydnar) [inline wit/testdata_test.go:507]: Can you pull this into a separate function, e.g. `TestHasBorrow` in abi_test.go?

**Review by ydnar — CHANGES_REQUESTED** [review]

File: wit/testdata_test.go (hunk 2 above)
Comment (Mossaka) [reply wit/testdata_test.go]: Done

**Review by ydnar — APPROVED** [review]

**Pull request merged**